rust/numaflow-monitor/src/runtime.rs

```diff
@@ -7,12 +7,20 @@ use std::convert::TryFrom;
 use std::fs;
 use std::fs::File;
 use std::io::Write;
+use std::os::unix::fs::DirBuilderExt as _;
 use std::path::Path;
 use std::str;
+use std::sync::{Arc, OnceLock};
+use tokio::sync::Mutex;
 use tonic::Status;
 use tracing::error;
+use uuid::Uuid;
 
-const CURRENT_FILE: &str = "current-numa.json";
+static PERSIST_ERROR_LOCK: OnceLock<Arc<Mutex<()>>> = OnceLock::new();
+
+fn get_persist_error_lock() -> &'static Arc<Mutex<()>> {
+    PERSIST_ERROR_LOCK.get_or_init(|| Arc::new(Mutex::new(())))
+}
```

Comment:
We don't need a lock, we only surface critical errors that caused the pod to crash or go to container state, we could make the `persist_application_error` method `FnOnce`.